reference.md

```diff
@@ -1005,6 +1002,30 @@ macros into smaller scopes as well. Note that macros are a
 compile-time construct; they do not exist at runtime. As such macros
 cannot be exported at the bottom of a module like functions and other values.
 
+Broadly, a macro replaces its own expression with another.
+
+Macros should return a "quoted" expression (via `` ` ``) to be used as the
+replacement. Normally that expression would "unquote" a value (via `,`) but it
+does not have to. You may not "unquote" a value outside of a "quoted"
+expression.
+
+As a very simple example, the following macro will emit an expression that adds
+10 to a given argument.
+
+```fennel
+(macro plus-10 [arg]
+  `(+ 10 ,arg))
+; ^ quote the following expression, it will be emitted in place of (plus-10 ...)
```

Comment:
I think emit is a normal enough term

Reply:
on a separate note, while this example shows how macros work, it doesn't show why they're necessary. I mean `plus-10` can be a function without a problem. And this also may encourage people to write macros where a function will do, which is an antipattern.

I think there was an example macro in the tutorial, so perhaps we could instead just link to it here?

Reply:
Yeah, in OP^

> Macro explanation uses plus-10, which might infer you should be replacing plus-10 functions with macros. Not sure if that needs a note or not.

I agree it is not "a good macro" but I think you want a example where working out what the macro does isn't difficult.

There isn't a an explicit `macro` in the (current?) tutorial on gh but the function could be used:

I think you really want a one liner though.

```
; from tutorial when ctrl-f macro
(fn print-and-add [a b c]
  (print a)
  (+ b c))
```

Reply:
I added a note that it's a poor macro, can't think of a simpler example that doesn't make you think about what the macro is doing while being better served as a function.

Reply:
> There isn't a an explicit macro in the (current?) tutorial on gh but the function could be used:

Sorry, there's a macro example in https://github.com/bakpakin/Fennel/blob/main/reference.md#import-macros-load-macros-from-a-separate-module, but we were talking with @technomancy on IRC about putting a bit more info to the tutorial